src/backend/distributed/master/master_node_protocol.c

```diff
@@ -217,8 +217,11 @@ master_get_table_ddl_events(PG_FUNCTION_ARGS)
 		MemoryContext oldContext = MemoryContextSwitchTo(
 			functionContext->multi_call_memory_ctx);
 
+		bool includeForeignTableDependencies = true;
+
 		/* allocate DDL statements, and then save position in DDL statements */
-		List *tableDDLEventList = GetTableDDLEvents(relationId, includeSequenceDefaults);
+		List *tableDDLEventList = GetTableDDLEvents(relationId, includeSequenceDefaults,
```

Comment:
I didn't have time to look in detail. But, do we strictly need the `includeForeignTableDependencies ` parameter? To me the name `GetTableDDLEvents ` slightly conflicts with the concept `foreign keys` because now we're talking about multiple tables. If possible, I'd be in favor of not adding the parameter.

Reply:
To be more clear, this parameter is about "foreign table dependencies", not about "foreign keys". This function, by default, was bringing dependencies for the foreign table too (such as foreign server).
As we are in the node that has those dependencies already, execution of those commands were erroring out. Maybe we should append "IF NOT EXISTS" to the end of extension & server creation commands generated in below ? 

https://github.com/citusdata/citus/blob/347597bebf2104a71425287dd9a86d5ba55773a9/src/backend/distributed/master/master_node_protocol.c#L616-L623

Reply:
If that sounds good, I can open a small pr to against master to do the proposed change ?

Reply:
Or, it might make more sense to setting `if_not_exist` flag for the commands to be executed in `CreateShellTableForCitusLocalTable` function

Reply:
Applied the change I mentioned above in https://github.com/citusdata/citus/pull/3852/commits/1a664d002c49a60e81421d8d42283a6e5d53e214, I can split this commit to another pr if you want